README.md

```diff
@@ -93,57 +93,126 @@ dicom-anonymizer InputFilePath OutputFilePath --dictionary dictionary.json
+    parser.add_argument('--suffix', action='store', help='Suffix that will be added at the end of series description')
+    args = parser.parse_args()
+
+    InputPath = args.input
+    OutputPath = args.output
+
+    dictionary = {}
+
+    def setupSeriesDescription(dataset, tag):
+        element = dataset.get(tag)
+        if element is not None:
+            element.value = element.value + '-' + args.suffix
+
+    # Store all tags into one array and setup the action in
+    # order to keep these value as it is
+    # All tags are defined on file dicomfields.py
+    allTags = []
+    allTags.extend(D_TAGS)
+    allTags.extend(Z_TAGS)
+    allTags.extend(X_TAGS)
+    allTags.extend(U_TAGS)
+    allTags.extend(Z_D_TAGS)
+    allTags.extend(X_Z_TAGS)
+    allTags.extend(X_D_TAGS)
+    allTags.extend(X_Z_D_TAGS)
+    allTags.extend(X_Z_U_STAR_TAGS)
+
+    # the 'keep' method is already defined into the dicom-anonymizer
+    # It will overrides the default behaviour
+    for i in allTags:
+        dictionary[i] = keep
+
+    if args.suffix:
+        dictionary[(0x0008, 0x103E)] = setupSeriesDescription
+
+    # Launch the anonymization
+    anonymize(InputPath, OutputPath, dictionary)
+
+if __name__ == "__main__":
+    main()
 ```
 
 In your own file, you'll have to define:
 - Your custom functions. Be careful, your functions always have in inputs a dataset and a tag
-- A dictionary which map your functions to a string
+- A dictionary which map your functions to a tag
+
+## Anonymize dicom tags without dicom file
 
-# Overrides actions for default tags
-You can have the need to keep some tags, for example UID. Then, instead of replace it (`replaceUID` action) then you can override it :
+For some reason, you need to anonymize dicom fields without initial dicom file (extracted from a database for example). Here is how you can do it:
+```python
+from dicomanonymizer import *
+
+def main():
+
+  # Create a list of tags object that should contains id, type and value
+  fieldsList = [
+    { # Replaced by Anonymized
+      'id': '(0x0040, 0xA123)',
+      'type': 'LO',
+      'value': 'Annie de la Fontaine',
+    },
+    { # Replaced with empty value
+      'id': '(0x0008, 0x0050)',
+      'type': 'TM',
+      'value': 'toto',
+    },
+    { # Deleted
+      'id': '(0x0018, 0x4000)',
+      'type': 'VR',
+      'value': 'toto',
+    }
+  ]
+
+  # Create a readable dataset for pydicom
+  data = pydicom.Dataset()
+
+  # Add each field into the dataset
+  for field in fieldsList:
+    tag = field['id']
+    if isinstance(tag, str):
+      # Transform string to hex values
+      splitTags = tag.replace('(', '').replace(')', '').replace(', ', ' ').split()
+      if len(splitTags) != 2:
+        continue
+      val1 = hex(int(splitTags[0], 16))
+      val2 = hex(int(splitTags[1], 16))
+      tag = (val1, val2)
+
+    data.add_new(tag, field['type'], field['value'])
+
+  anonymizeDataset(data)
```

Comment:
Pydicom.dataset derives from a dict. Can't you use only the dict API in your anonymizer ?

Reply:
pydicom.Dataset derives from dict but has more methods that are used in the anonymizer (walk). We have to use this class.

Reply:
Understood.
But to me this constraint should not be on the user side.

maybe you can, in anonymizeDataset, check the input type of data and if it is a dict, wrap it into a pydicom.dataset

Reply:
I prefer to let the user create its own dataset because:
- Using a pydicom.dataset is the more generic way to not making the dicomanonymizer failed or adding extra text to recreate a Dataset
- We don't want that the dicomanonymizer manages the input data because the probability to having a bad map ids or anything else is increased. Imposing a Dataset seems more well-defined and avoird a lot of inconvenience on dicomanonymizer's side
- The user can manage also the private elements, private blocks, sequences,...,

Reply:
fair enough, you can leave it as is.

you might want to rename `data` into `dataset` to give a clue that it is a pydicom.Dataset.